bson-adt-mongo3-async/src/main/scala/adt/bson/mongo/FunctionImplicits.scala

```diff
@@ -0,0 +1,51 @@
+package adt.bson.mongo
+
+import com.mongodb.async.SingleResultCallback
+import play.api.libs.functional.ContravariantFunctor
+
+import scala.concurrent.Promise
+import scala.language.implicitConversions
+import scala.util.{Failure, Success, Try}
+
+// TODO: Avoid this redundant trait by moving the common mongo-core-driver dependent code to a shared module
```

Comment:
This is my next step for the project. I guess I ought to make a ticket in github for this.